http/client.go

```diff
@@ -14,46 +14,92 @@ import (
 	"time"
 )
 
-// New creates a new http.Client with a custom certificate, which can be loaded from the passed CA Bundle file and/or
-// directory. If both CABundleFile and CABundleDir are empty arguments, it creates an unsecure HTTP client.
-func New(CABundleFile, CABundleDir string, httpTimeout time.Duration) (*http.Client, error) {
-	return _new(CABundleFile, CABundleDir, httpTimeout, "")
-}
+type ClientOption func(*http.Client) error
 
-// NewAcceptInvalidHostname new http.Client with ability to accept HTTPS certificates that don't
-// match the hostname of the server they are connecting to.
-func NewAcceptInvalidHostname(CABundleFile, CABundleDir string, httpTimeout time.Duration, hostname string) (*http.Client, error) {
-	return _new(CABundleFile, CABundleDir, httpTimeout, hostname)
-}
+// New creates a new http.Client with the Passed Client Options
+// that will have a custom certificate if it is loaded from the passed CA Bundle file and/or
+// directory. If both CABundleFile and CABundleDir are empty arguments, it creates an unsecure HTTP client.
+func New(opts ...ClientOption) (*http.Client, error) {
+	t := &http.Transport{
+		TLSClientConfig: &tls.Config{},
+	}
 
-func _new(CABundleFile, CABundleDir string, httpTimeout time.Duration, acceptInvalidHostname string) (*http.Client, error) {
-	// go default http transport settings
-	t := &http.Transport{}
+	client := &http.Client{
+		Transport: t,
+	}
 
-	if CABundleFile != "" || CABundleDir != "" {
-		certs, err := getCertPool(CABundleFile, CABundleDir)
-		if err != nil {
+	for _, opt := range opts {
+		if err := opt(client); err != nil {
 			return nil, err
 		}
+	}
+
+	return client, nil
+}
+
+// WithCABundleFile adds the CABundleFile cert to the the client's certPool,
+// if a CABundleDir is passed it will be joined to the CABundleFile to try to get the file.
+// If the file is a full path, even it the CABundleDir is passed, it will be detected and the full
+// path will be used instead
+func WithCABundleFile(CABundleFile, CABundleDir string) ClientOption {
```

Comment:
why don't we just remove the dir option and require file to be the path?

Reply:
I put it bellow, I agree, but we could be breaking the behaviour for some users.

Reply:
This is already a breaking change as we are modifying the constructor. So, we/sdk-users would notice about this anyway and would need to adjust it accordingly either way.

Reply:
Ok, modified!